src/modules/Utilities/private/Invoke-WebRequestWithRetry.ps1

```diff
@@ -0,0 +1,103 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+function Invoke-WebRequestWithRetry {
+    param(
+        [Parameter(Mandatory = $true)]
+        [System.Uri]$Uri,
+
+        [Parameter(Mandatory = $false)]
+        [Microsoft.PowerShell.Commands.WebRequestMethod]$Method = 'Get',
+
+        [Parameter(Mandatory = $false)]
+        [System.Collections.IDictionary]$Headers,
+
+        [Parameter (Mandatory = $false)]
+        [System.String]$ContentType,
+
+        [Parameter(Mandatory = $false)]
+        [System.Object] $Body,
+
+        [Parameter(Mandatory = $false)]
+        [Switch] $DisableKeepAlive,
+
+        [Parameter(Mandatory = $false)]
+        [Switch] $UseBasicParsing,
+
+        [Parameter(Mandatory = $false)]
+        [System.Management.Automation.PSCredential]
+        [System.Management.Automation.Credential()]$Credential,
+
+        [Parameter(Mandatory = $false)]
+        [int]$TimeoutInSec = 600,
+
+        [Parameter(Mandatory = $false)]
+        [Switch]$Retry
+    )
+
+    $params = @{
+        'Headers'     = $Headers;
+        'ContentType' = $ContentType;
+        'Method'      = $Method;
+        'Uri'         = $Uri;
+        'TimeoutSec'  = $TimeoutInSec
+    }
+
+    if ($null -ne $Body) {
+        $params.Add('Body', $Body)
+    }
+
+    if ($DisableKeepAlive.IsPresent) {
+        $params.Add('DisableKeepAlive', $true)
+    }
+
+    if ($UseBasicParsing.IsPresent) {
+        $params.Add('UseBasicParsing', $true)
+    }
+
+    if ($Credential -ne [System.Management.Automation.PSCredential]::Empty -and $null -ne $Credential) {
+        $params.Add('Credential', $Credential)
+    }
+    else {
+        $params.Add('UseDefaultCredentials', $true)
+    }
+
+    $retryIntervalInSeconds = 10
```

Comment:
Would it be useful if we make RetryInterval and MaxRetry as non-mandatory parameters with default value?

Reply:
Added `-MaxRetry` and `-RetryIntervalInSeconds` as parameters for both Invoke methods.